### PR TITLE
Fix NestedBlockDepth false negative

### LINKED ISSUE #2085: NestedBlockDepth false negative
NestedBlockDepth should find an issue here:
```kotlin
    fun procedure(n: Int) {
        if (true) {
            if (true) {
            } else if (true) {
                if (true) {
                    if (true) {
                        if (true) { 
                            if (true) { 
                                if (true) {
                                }
                            }
                        }
                    }
                }
            }
        }
    }
```

But it doesn't. It seems that the `if else` is breaking something.

### PR DESCRIPTION
**Pull request by BraisGabin**
Fix #2085

I also improve the tests a bit simplifing the code and using `compileAndLint()`

### CODE REVIEW

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NestedBlockDepthSpec.kt [hunk 1]
```diff
@@ -52,27 +52,33 @@ class NestedBlockDepthSpec : Spek({
         }
 
         it("should not count else if as two") {
```
Comment (schalkms) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NestedBlockDepthSpec.kt]: The test description feels hard to read. Maybe we can find a better name here.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NestedBlockDepthSpec.kt [hunk 2]
```diff
@@ -52,27 +52,33 @@ class NestedBlockDepthSpec : Spek({
...
-                    deleteOneOrTwoSemicolons(node as LeafPsiElement)
-                }
-            } else if (psi.isSemicolon()) {
-                val nextLeaf = psi.nextLeaf()
-                if (nextLeaf.isSemicolonOrEOF() || nextTokenHasSpaces(nextLeaf)) {
-                    addFindings(CodeSmell(id, Entity.from(psi)))
-                    withAutoCorrect { psi.delete() }
-                }
-            }
+        it("should not count else if as two but it should count it") {
```
Comment (schalkms) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NestedBlockDepthSpec.kt]: The same holds true for this test case description.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NestedBlockDepthSpec.kt (hunk 1 above)
Comment (BraisGabin) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NestedBlockDepthSpec.kt]: Any idea? This test name was there before my change...

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NestedBlockDepthSpec.kt (hunk 2 above)
Comment (schalkms) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NestedBlockDepthSpec.kt]:
```suggestion
        it("reports deeply nested if else branches") {
```

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NestedBlockDepthSpec.kt (hunk 1 above)
Comment (schalkms) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NestedBlockDepthSpec.kt]:
```suggestion
        it("does not report valid nested if else branches") {
```

**Comment by schalkms** [comment]
@BraisGabin please commit my test case naming suggestions from the code review, if you are okay with those.

**Comment by arturbosch** [comment]
Thanks again for the great PR.It kinda feels wrong to hold back all the good improvements and fixes from the community :/. I try to finish https://github.com/arturbosch/detekt/issues/1921 this WE and release a new version soon.

**Comment by BraisGabin** [comment]
Done. Sorry for the late update. Busy days.

**Pull request merged**